tony-core/src/main/java/com/linkedin/tony/models/JobMetadata.java

```diff
@@ -20,29 +21,88 @@
   private String status;
   private String user;
 
-  public JobMetadata(String id, String jobLink, String configLink, String rmLink, long started, long completed, String status, String user) {
-    this.id = id;
-    this.jobLink = jobLink;
-    this.configLink = configLink;
-    this.rmLink = rmLink;
-    this.started = started;
-    this.completed = completed;
-    this.status = status;
-    this.user = user;
+  private JobMetadata(JobMetadata.Builder builder) {
+    this.id = builder.id;
+    this.jobLink = "/" + Constants.JOBS_SUFFIX + "/" + id;
+    this.configLink = "/" + Constants.CONFIG_SUFFIX + "/" + id;
+    this.rmLink = Utils.buildRMUrl(builder.conf, builder.id);
+    this.started = builder.started;
+    this.completed = builder.completed;
+    this.status = builder.status;
+    this.user = builder.user;
+  }
+
+  // for testing only
+  public JobMetadata() {
```

Comment:
You can add `@VisibleForTesting` so IDE would WARN if we call it from code. But why don't we use Builder in test as well to create a JobMetadata instance?

Reply:
Since there's a Builder, I think we should make it the only way of creating JobMetadata instances (we should strive for consistent APIs everywhere). So make all the JobMetadata constructors private and only make JobMetadata.Builder public.

Reply:
👍